stream.go

```diff
@@ -101,7 +101,7 @@ type ClientStream interface {
 }
 
 // NewStream creates a new Stream for the client side. This is typically
-// called by generated code.
+// called by generated code. ctx is used for the lifetime of the stream.
```

Comment:
If you're making changes here, would you mind summarizing what I wrote in #1854 regarding how to ensure a stream doesn't leak a goroutine?  Something like:

```go
// To ensure resources are not leaked due to the stream returned, one of the following
// actions must be performed:
//
// 1. call Close on the ClientConn,
// 2. cancel the context provided,
// 3. call RecvMsg until a non-nil error is returned, or
// 4. receive a non-nil error from Header or SendMsg besides io.EOF.
//
// If none of the above happen, a goroutine and a context will be leaked, and grpc
// will not call the optionally-configured stats handler with a stats.End message.
```

Reply:
Done